pkg/cmd/pulumi/state/state_reprotect.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package state
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/backend"
+	"github.com/pulumi/pulumi/pkg/v3/resource/deploy"
+	"github.com/pulumi/pulumi/pkg/v3/resource/edit"
+	pkgWorkspace "github.com/pulumi/pulumi/pkg/v3/workspace"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+
+	"github.com/spf13/cobra"
+)
+
+func newStateReprotectCommand() *cobra.Command {
+	var protectAll bool
+	var stack string
+	var yes bool
+
+	cmd := &cobra.Command{
+		Use:   "reprotect [resource URN]",
+		Short: "Re-protect (undo a `pulumi unprotect`) resources in a stack's state",
+		Long: `Re-Protect resource in a stack's state
+
+This command sets the 'protect' bit on one or more resources, preventing those resources to be deleted.
+
+Caution: This is will not protect resources that were not marked as ` + "`protect`" + ` in the program.
+In the case that the resource is not protected in the program, the next ` + "`pulumi up`" + ` would 
+overwrite the protection (eg when you deploy via pulumi deployments). Instead use this only to re-enable
+protect on resources which were unprotected with ` + "`pulumi unprotect`" + `.
```

Comment:
This isn't entirely accurate -- it will protect against deletions by e.g. `pulumi destroy`. I think we should probably just remove this help text, given that this is already a power tool.

Reply:
Now I'm curious how this behaves with `pulumi destroy --run-program=true`. Presumably it uses what the program specifies (would need to verify). If so, when running the program becomes the default behavior for destory, you could run `pulumi state protect`, forget to update the program, then run `pulumi destroy` (with run program as default) and it will delete the resource you just protected via the command line.